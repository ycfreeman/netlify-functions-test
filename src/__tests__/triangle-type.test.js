/* eslint-disable no-await-in-loop */
const axios = require("axios");

describe("triangle-type", () => {
  const api = axios.create({
    // baseURL: "https://knockknock.readify.net/api"
    baseURL: "http://localhost/.netlify/functions/api"
  });

  const url = "/triangletype";

  it("success", async () => {
    const { data } = await api.request({
      url,
      params: { a: 3, b: 4, c: 5 }
    });
    expect(data).toMatchSnapshot();
    // expect(headers).toMatchSnapshot();
  });

  it("text/json", async () => {
    const { data } = await api.request({
      url,
      params: { a: 3, b: 4, c: 5 }
    });
    expect(data).toMatchSnapshot();
    // expect(headers).toMatchSnapshot();
  });

  it("methods", async done => {
    await Promise.all(
      ["get", "head", "post", "put", "delete", "options", "patch"].map(
        async method => {
          try {
            const { data } = await api.request({
              url,
              params: { a: 3, b: 4, c: 5 },
              method
            });
            expect(data).toMatchSnapshot(method);
          } catch (e) {
            expect(e.response.data).toMatchSnapshot(method);
            expect(e).toMatchSnapshot(method);
          }
        }
      )
    );

    done();
  });

  it("missing param", async () => {
    try {
      const { data } = await api.request({
        url
      });
      expect(data).toMatchSnapshot();
    } catch (e) {
      // expect(e.response.data).toMatchSnapshot();
      expect(e).toMatchSnapshot();
    }

    try {
      const { data } = await api.request({
        url,
        params: { a: 3, b: 4 }
      });
      expect(data).toMatchSnapshot();
    } catch (e) {
      // expect(e.response.data).toMatchSnapshot();
      expect(e).toMatchSnapshot();
    }

    try {
      const { data } = await api.request({
        url,
        params: { a: 3 }
      });
      expect(data).toMatchSnapshot();
    } catch (e) {
      // expect(e.response.data).toMatchSnapshot();
      expect(e).toMatchSnapshot();
    }
  });

  it("string", async () => {
    try {
      const { data } = await api.request({
        url,
        params: { a: "a", b: "b", c: 5 }
      });
      expect(data).toMatchSnapshot();
    } catch (e) {
      expect(e.response.data).toMatchSnapshot();
      expect(e).toMatchSnapshot();
    }
  });

  it("fraction", async () => {
    try {
      const { data } = await api.request({
        url,
        params: { a: 3.3, b: 4.4, c: 5.5 }
      });
      expect(data).toMatchSnapshot();
    } catch (e) {
      expect(e.response.data).toMatchSnapshot();
      expect(e).toMatchSnapshot();
    }
  });

  it("different types", async () => {
    await Promise.all(
      [[3, 4, 5], [3, 3, 3], [4, 4, 5], [1, 2, 4], [0, 0, 0]].map(
        async ([a, b, c]) => {
          try {
            const { data } = await api.request({
              url,
              params: { a, b, c }
            });
            expect(data).toMatchSnapshot(`${a}, ${b}, ${c}`);
          } catch (e) {
            expect(e.response.data).toMatchSnapshot(`${a}, ${b}, ${c}`);
            expect(e).toMatchSnapshot(`${a}, ${b}, ${c}`);
          }
        }
      )
    );
  });

  it("limit?", async done => {
    let error;
    let s = [3, 4, 5];

    do {
      try {
        s = s.map(x => x * 10);
        const [a, b, c] = s;

        const { data } = await api.request({
          url,
          params: {
            a,
            b,
            c
          }
        });
        expect(data).toMatchSnapshot(`${a}, ${b}, ${c}`);
      } catch (e) {
        error = e;
      }
    } while (!error);

    expect(`loop stops at ${s.join(", ")}`).toMatchSnapshot();
    expect(error).toMatchSnapshot();
    expect(error.response.data).toMatchSnapshot();
    expect.hasAssertions();

    done();
  }, 1000000);
});
