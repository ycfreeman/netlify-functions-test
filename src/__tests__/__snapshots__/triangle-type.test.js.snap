// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`triangle-type different types: 0, 0, 0 1`] = `"Error"`;

exports[`triangle-type different types: 1, 2, 4 1`] = `"Error"`;

exports[`triangle-type different types: 3, 3, 3 1`] = `"Equilateral"`;

exports[`triangle-type different types: 3, 4, 5 1`] = `"Scalene"`;

exports[`triangle-type different types: 4, 4, 5 1`] = `"Isosceles"`;

exports[`triangle-type fraction 1`] = `
Object {
  "message": "The request is invalid.",
}
`;

exports[`triangle-type fraction 2`] = `[Error: Request failed with status code 400]`;

exports[`triangle-type limit? 1`] = `"loop stops at 3000000000, 4000000000, 5000000000"`;

exports[`triangle-type limit? 2`] = `[Error: Request failed with status code 400]`;

exports[`triangle-type limit? 3`] = `
Object {
  "message": "The request is invalid.",
}
`;

exports[`triangle-type limit?: 30, 40, 50 1`] = `"Scalene"`;

exports[`triangle-type limit?: 300, 400, 500 1`] = `"Scalene"`;

exports[`triangle-type limit?: 3000, 4000, 5000 1`] = `"Scalene"`;

exports[`triangle-type limit?: 30000, 40000, 50000 1`] = `"Scalene"`;

exports[`triangle-type limit?: 300000, 400000, 500000 1`] = `"Scalene"`;

exports[`triangle-type limit?: 3000000, 4000000, 5000000 1`] = `"Scalene"`;

exports[`triangle-type limit?: 30000000, 40000000, 50000000 1`] = `"Scalene"`;

exports[`triangle-type limit?: 300000000, 400000000, 500000000 1`] = `"Scalene"`;

exports[`triangle-type methods: delete 1`] = `
Object {
  "message": "The requested resource does not support http method 'DELETE'.",
}
`;

exports[`triangle-type methods: delete 2`] = `[Error: Request failed with status code 405]`;

exports[`triangle-type methods: get 1`] = `"Scalene"`;

exports[`triangle-type methods: head 1`] = `""`;

exports[`triangle-type methods: head 2`] = `[Error: Request failed with status code 405]`;

exports[`triangle-type methods: options 1`] = `
Object {
  "message": "The requested resource does not support http method 'OPTIONS'.",
}
`;

exports[`triangle-type methods: options 2`] = `[Error: Request failed with status code 405]`;

exports[`triangle-type methods: patch 1`] = `
Object {
  "message": "The requested resource does not support http method 'PATCH'.",
}
`;

exports[`triangle-type methods: patch 2`] = `[Error: Request failed with status code 405]`;

exports[`triangle-type methods: post 1`] = `
Object {
  "message": "The requested resource does not support http method 'POST'.",
}
`;

exports[`triangle-type methods: post 2`] = `[Error: Request failed with status code 405]`;

exports[`triangle-type methods: put 1`] = `
Object {
  "message": "The requested resource does not support http method 'PUT'.",
}
`;

exports[`triangle-type methods: put 2`] = `[Error: Request failed with status code 405]`;

exports[`triangle-type missing param 1`] = `[Error: Request failed with status code 404]`;

exports[`triangle-type missing param 2`] = `[Error: Request failed with status code 404]`;

exports[`triangle-type missing param 3`] = `[Error: Request failed with status code 404]`;

exports[`triangle-type string 1`] = `
Object {
  "message": "The request is invalid.",
}
`;

exports[`triangle-type string 2`] = `[Error: Request failed with status code 400]`;

exports[`triangle-type success 1`] = `"Scalene"`;

exports[`triangle-type text/json 1`] = `"Scalene"`;
